orbit/models/dlt.py

```diff
@@ -59,6 +62,26 @@ def _set_static_data_attributes(self):
         super()._set_static_data_attributes()
         self._set_global_trend_attributes()
 
+    # override regression penalty validation from LGT
+    def _set_regression_penalty(self):
```

Comment:
Let's make an independent method and call it something like `_validate_horshoe_penalty_size` and move all the validation logic there. Please also write clear docs on what it's intended to do.

Then you can override `_set_regression_penalty` to call the validation method and then call `super()._set_regression_penalty()`